### PR TITLE
Added type definitions for postgraphile

### PR DESCRIPTION
**Pull request by jobergner**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not provide its own types, and you can not add them.
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/postgraphile/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+// Type definitions for postgraphile 2.3
```
Comment (plantain-00) [inline types/postgraphile/index.d.ts]: it seems to be v4: 4.0.0-alpha2.30 https://github.com/postgraphql/postgraphql/blob/postgraphile/package.json

**Review by plantain-00 — CHANGES_REQUESTED** [review]

**Comment by Java-Jonas** [comment]
Hey,
thanks for the feedback. Unfortunately lint won't let me define the version like that.
(https://github.com/Microsoft/dtslint/blob/master/docs/dt-header.md)
Any suggestions for an alternative?

**Comment by plantain-00** [comment]
@andy-ms

**Comment by andy-ms** [comment]
You can write `// Type definitions for postgraphile 4.0`.

**Comment by Java-Jonas** [comment]
Sorry for the late response. Fixed it as mentioned by the contributor. 
Thanks for the Help!

**Review by plantain-00 — APPROVED** [review]

File: types/postgraphile/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+// Type definitions for postgraphile 4.0
+// Project: https://github.com/postgraphql/postgraphql/tree/postgraphile
```
Comment (andy-ms) [inline types/postgraphile/index.d.ts]: The url says postgraphile, but the README.md there says to `npm install -g postgraphql@next`. We should match the package name from NPM, so this directory should probably be named `postgraphql`.

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed review comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**